extends Node;

const hours: Array[String] = ["23:00", "23:30", "00:00", "00:30", "1:00", "1:30", "2:00", "2:30", "3:00", "3:30", "4:00", "4:30", "5:00", "5:30"];
const weights: Dictionary = {"ðŸŽ": 1.0, "ðŸŒ": 1.0, "ðŸ§„": 0.1, "ðŸž": 1.0, "ðŸ§€": 1.5, "ðŸ¥š": 1.5, "ðŸ§‡": 2.0, "ðŸ–": 3.0, "ðŸ•": 3.0, "ðŸ¦": 3.0, "ðŸ°": 4.0, "ðŸš": 4.0, "ðŸ’": 5.0, "ðŸŒ½": 6.0, "ðŸ¤": 7.0, "ðŸ¥ž": 8.0, "ðŸ«‘": 9.0, "ðŸŒ¶": 0.5, "ðŸº": 0.5, "ðŸ¥¦": 1.0, "ðŸ¥“": 0.1, "ðŸ‹â€ðŸŸ©": 1.0, "ðŸ«›": 0.5, "ðŸŒ®": 3.0, "ðŸ¥›": 1.5, "ðŸ¯": 2.5, "ðŸ¹": 0.5, "ðŸ¶": 1.5, "ðŸ«š": 2.0, "ðŸµ": 1.0, "â˜•ï¸": 1.0, "ðŸ«˜": 2.0, "ðŸ¥’": 1.0, "ðŸ¨": 1.5};
const time_speed: int = 1;

var actual_day: int = 1;
var time: int = 0;
var time_sum: float = 0.0;
var hour_text: String;
var halt_game: bool = true;

func _process(delta: float) -> void:
	if (not halt_game):
		time_sum += delta;
		if(time_sum >= time_speed):
			if (time < len(hours) - 1):
				time += 1;
				time_sum = 0.0;
				format_hour();
			else:
				halt_game = true;
				check_game_progress();
		
func format_hour() -> void:
	hour_text = hours[time];
	Events.emit_signal("update_time");

func check_game_progress() -> void:
	if (Player.hungry <= 9.0):
		get_tree().change_scene_to_file("res://Scenes/end_day.tscn");
	else:
		get_tree().change_scene_to_file("res://Scenes/game_over.tscn");

func new_day() -> void:
	actual_day += 1;
	time = 0;
	time_sum = 0.0;
	Events.emit_signal("update_day");
	format_hour();
	Player._another_day();
	# other stuff to change for the player and the game after the day passes
	# check if your hunger was filled;

func restart_game() -> void:
	actual_day = 1;
	time = 0;
	time_sum = 0.0;
	hour_text = hours[time];
